rplidar.go

```diff
@@ -311,3 +335,83 @@ func pointFrom(yaw, pitch, distance float64, reflectivity uint8) (r3.Vector, poi
 
 	return pos, d
 }
+
+// checkLockFiles compares the current process and device_path to rplidar.lock files to see if any ongoing
+// sessions for that device path still exist
+func checkLockFiles(devicePath string) (string, error) {
+
+	// Get rplidar related processes
+	rplidarProcesses, err := getRplidarProcesses()
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting rplidar-module related processes")
+	}
+	currentProcess := rplidarProcesses[len(rplidarProcesses)-1]
+	oldProcesses := rplidarProcesses[:len(rplidarProcesses)-1]
+
+	// Get rplidar related lock files
+	files, err := os.ReadDir(rplidarModuleLockDir)
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading lock file directory")
+	}
+	var rplidarLockFiles []string
+	for _, file := range files {
+		if strings.Contains(file.Name(), "rplidar") {
+			rplidarLockFiles = append(rplidarLockFiles, file.Name())
+		}
+	}
+
+	// Look through lock fils for those relating to active processes + given device path, if a lock file refers to
+	// a no longer active process, delete it
+	for _, lockFileName := range rplidarLockFiles {
+		var matchFound bool
+		for _, oldProc := range oldProcesses {
+			if strings.Contains(lockFileName, fmt.Sprintf("pid%v", oldProc)) {
+				matchFound = true
+				if strings.Contains(lockFileName, fmt.Sprintf("dv%v", devicePath[devicePathPrefixOffset:])) {
+					return "", errors.Errorf("another rplidar-module process(s) using the same serial_path has been found, "+
```

Comment:
[q] `process(s)`?

Reply:
That is an artifact from when I was debating returning all processes